pkg/cloudprovider/vsphereparavirtual/route.go

```diff
@@ -0,0 +1,415 @@
+	if err != nil {
+		return fmt.Errorf("error listing nodes: %v", err)
+	}
+	if len(nodes) == 0 {
+		klog.V(6).Infof("No nodes available")
+		return nil
+	}
+
+	for _, node := range nodes {
+		if node.Name == nodeName {
+			_, condition := cloudnodeutil.GetNodeCondition(&(node.Status), v1.NodeNetworkUnavailable)
+			if condition != nil && condition.Status == v1.ConditionTrue {
+				klog.V(6).Infof("removing node condition %v:%+v because the Route CR is successfully created", nodeName, condition)
+
+				err = clientretry.RetryOnConflict(updateNetworkConditionBackoff, func() error {
+					var err error
+					currentTime := metav1.Now()
+					err = cloudnodeutil.SetNodeCondition(r.kubeClient, types.NodeName(nodeName), v1.NodeCondition{
+						Type:               v1.NodeNetworkUnavailable,
+						Status:             v1.ConditionFalse,
+						Reason:             "RouteCreated",
+						Message:            "Route CR created",
+						LastTransitionTime: currentTime,
+					})
+					if err != nil {
+						klog.V(6).Infof("Error updating node %s, retrying: %v", types.NodeName(node.Name), err)
+					}
+					return err
+				})
+
+				if err != nil {
+					klog.Errorf("Error updating node %s: %v", nodeName, err)
+				}
+			}
+		}
+	}
+
+	klog.V(6).Infof("Successfully updated NetworkingCondition for node %s", nodeName)
+	return err
+}
+
+// checkStaticRouteRealizedState checks static route realized state
+// The check happens every 1 second and the default timeout is 10 seconds
+// func (r *routesProvider) checkStaticRouteRealizedState(routeSetName string) error {
+// 	timeout := time.After(RealizedStateTimeout)
+// 	ticker := time.NewTicker(RealizedStateSleepTime)
+// 	defer ticker.Stop()
+// 	for {
+// 		select {
+// 		case <-timeout:
+// 			return fmt.Errorf("timed out waiting for static route %s", routeSetName)
+// 		case <-ticker.C:
+// 			routeSet, err := r.routeClient.NsxV1alpha1().RouteSets(r.namespace).Get(context.Background(), routeSetName, metav1.GetOptions{})
+// 			if err != nil {
+// 				klog.ErrorS(ErrListRouteSet, fmt.Sprintf("%v", err))
+// 				return err
+// 			}
+// 			condition := GetRouteSetCondition(&(routeSet.Status), v1alpha1.RouteSetConditionTypeReady)
+// 			if condition != nil && condition.Status == v1.ConditionTrue {
+// 				return nil
+// 			}
+// 		}
+// 	}
+// }
+
+// DeleteRoute implements Routes.DeleteRoute
+// Delete node's corresponding RouteSet CR
+func (r *routesProvider) DeleteRoute(ctx context.Context, clusterName string, route *cloudprovider.Route) error {
+	routeSetName := string(route.TargetNode)
+	klog.V(6).Infof("Deleting RouteSet CR %s in cluster %s", routeSetName, clusterName)
+	return r.DeleteRouteSetCR(routeSetName)
+}
+
+// GetRouteName returns Route name as <nodeName>-<cidr>-<clusterName>
+// e.g. nodeName-100.96.0.0-24-clusterName
+func (r *routesProvider) GetRouteName(nodeName string, cidr string, clusterName string) string {
+	return strings.Replace(nodeName+"-"+cidr+"-"+clusterName, "/", "-", -1)
```

Comment:
Is it better to put `clusterName` in the front?

Reply:
Because can't end the string with `.` or number, that's why the cluster name is put at the end

Reply:
yeah, we can only change it to something like `<clusterName>-<cidr>-<nodeName>`, but I think clusterName is only added for passing format validation, and nodeName should be put first as it maps to the corresponding route

Reply:
Got it, thanks